### PR TITLE
 reconcile gitlab ignore group bots

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**
as we observe group tokens breaing reconcile,
i started a experiment with filtering bot usernames

the correct solution is to eventually update python-gitlab
 but the breaking api changes needed more time than i have right now

### CODE REVIEW

File: reconcile/utils/gitlab_api.py [hunk 1]
```diff
@@ -227,14 +230,27 @@ def get_app_sre_group_users(self):
...
+        access_level: str
+
+    @staticmethod
+    def _is_bot_username(username: str) -> bool:
+        """crudely checking for the username
+
+        as gitlab-python require a major upgrade to use the billable members apis
+
+        """
+        return username.startswith("group_") and "_bot_" in username
```
[the remark this answers is not in the document]
Comment (RonnyPfannschmidt) [reply reconcile/utils/gitlab_api.py:251]: @hemslo it seems there is no base example of a test that's using a instance of GitlabApi - it seems that was completely skipped

[the remark this answers is not in the document]
Comment (hemslo) [reply reconcile/utils/gitlab_api.py:251]: Maybe need a new test case in https://github.com/app-sre/qontract-reconcile/blob/96af9d75a4e779591442ecce236f9835754fbf1e/reconcile/test/utils/test_gitlab_api.py

File: reconcile/utils/gitlab_api.py [hunk 2]
```diff
@@ -82,7 +85,7 @@ def __init__(
         if not secret_reader:
             secret_reader = SecretReader(settings=settings)
         token = secret_reader.read(instance["token"])
-        ssl_verify = instance["sslVerify"]
+        ssl_verify = instance["sslVerify"] if ssl_verify is None else ssl_verify
```
[the remark this answers is not in the document]
Comment (RonnyPfannschmidt) [reply reconcile/utils/gitlab_api.py]: done

File: reconcile/utils/gitlab_api.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (RonnyPfannschmidt) [reply reconcile/utils/gitlab_api.py]: @hemslo tests are implemented, the test refactoring is a bit larger than anticipated

moving the creation of the gitlab api instance to a fixture created a need to put the mocks on it via a logic move

i can migrate the creation of the GitlabApi instances from having a smart ctor over to using a named ctor + plain old construction to make this easier

[the remark this answers is not in the document]
Comment (hemslo) [reply reconcile/utils/gitlab_api.py]: Looks nice! Thank you!

File: reconcile/test/test_ocm_machine_pools.py [hunk 3]
```diff
@@ -33,6 +33,7 @@
 
 
 class TestPool(AbstractPool):
+    __test__ = False
```
Comment (lechuk47) [inline reconcile/test/test_ocm_machine_pools.py:36]: Remove this please, we'll handle it in a different PR

File: reconcile/test/test_status_board.py [hunk 4]
```diff
@@ -19,6 +19,7 @@
 
 
 class TestStatusBoard(AbstractStatusBoard):
+    __test__ = False
```
Comment (lechuk47) [inline reconcile/test/test_status_board.py:22]: Remove this please, we'll handle it in a different PR

Comment (RonnyPfannschmidt) [reply reconcile/test/test_status_board.py:22]: dito

File: reconcile/test/test_ocm_machine_pools.py (hunk 3 above)
Comment (RonnyPfannschmidt) [reply reconcile/test/test_ocm_machine_pools.py:36]: https://github.com/app-sre/qontract-reconcile/pull/4041

**Comment by RonnyPfannschmidt** [comment]
moved the __test__ changes to https://github.com/app-sre/qontract-reconcile/pull/4041

**Pull request merged**